### ISSUE TITLE
Standalone app as fallback for when Crankshaft breaks

### ISSUE THREAD

**Issue by aarron-lee** [ShadowBlip/HandyPT#21]
Whenever Crankshaft breaks due to gamepadui updates, it cripples devices that need the TDP controls.

Is it possible to somehow wrap HandyPt up into a separate app that we can add to steam via the `add non-steam game` route? That'd be a useful fallback to have for whenever Crankshaft breaks due to gamepadui updates.

This would also allow handyPT to be used on generic linux distros and desktops, as opposed to being tightly coupled with gamepadui and crankshaft.

I currently am working around the crankshaft issue by manually running console commands with ryzenadj, but it's not the greatest user experience, and having to do this every time crankshaft breaks is not ideal.

**Comment by pastaq** [comment]
Oh I feel you. I'm in the same boat right now and I spent countless hours making the damn thing. Unfortunately thats not really possible without a full rewrite. HandyPT is essentially jsut a crankshaft front end for RyzenAdj. I'm working with the PowerTools dev to bring this funcitonality to decky-loader right now at least. It also tends to break about as often, but at least has more active development, so hopefully that will be a bit better in the short term. In the long term we're working on an open source independant overlay for this that valve/steam can't break.

**Comment by pastaq** [comment]
In the mean time, you can add the bash commands for RyzenAdj to each game. That will at least allow per-game profiles without going to the CLI every time.

Example:
https://www.reddit.com/r/SteamPlay/comments/ajg7b9/launch_a_bash_script_beforeafter_a_game_runs/?utm_source=share&utm_medium=android_app&utm_name=androidcss&utm_term=1&utm_content=share_button

**Comment by aarron-lee** [comment]
> Unfortunately thats not really possible without a full rewrite. HandyPT is essentially jsut a crankshaft front end for RyzenAdj.

Gotcha, if that's the case, closing this github issue for now then.

**Comment by aarron-lee** [comment]
Just wanted to followup on this, I ended up creating a very barebones TDP AppImage ([link](https://github.com/aarron-lee/simple-ryzen-tdp/) ) that I've started using on the Aya Neo Air Pro for TDP control. I'm reusing the `ryzenadj` binary in the handypt app, and using the same sudoers workaround for passwordless sudo.

I needed to do a `frzr-unlock` and install fuse2 (`sudo pacman -S fuse2`). The AppImage has been working well as an emergency backup option on the Aya Neo Air. I used [appImageLauncher](https://github.com/TheAssassin/AppImageLauncher) to generate a `.desktop` entry for the AppImage while on the Gnome desktop, and then added the tdp conrol app to steam as a non-steam game.
